core/src/main/java/jenkins/model/experimentalflags/BooleanUserExperimentalFlag.java

```diff
@@ -36,7 +37,7 @@ protected BooleanUserExperimentalFlag(@NonNull String flagKey) {
 
     @Override
     public @NonNull Boolean getDefaultValue() {
-        return false;
+        return SystemProperties.getBoolean(getClass().getName() + ".defaultValue");
```

Comment:
seems like an implementation detail that its the class name, wouldn't it be better to use the key?
```suggestion
        return SystemProperties.getBoolean(getFlagKey() + ".defaultValue");
```